### ISSUE TITLE
Code formatting introduces 'max-len' error.

### ISSUE THREAD

**Issue by rishabh-42** [eslint/eslint#15028]
**Tell us about your environment**

* **ESLint Version:**  5.16.0
* **Node Version:** 8.16.0
* **npm Version:** 6.4.1
* **Operating System:** Linux (Ubuntu 18.04)

**What parser (default, `@babel/eslint-parser`, `@typescript-eslint/parser`, etc.) are you using?**
**default**

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
/**
 * ESLint config file
 */
module.exports = { extends: 'airbnb-base', rules: { 'no-throw-literal': 'off' } };
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**
I have written a piece of code:  (badCode.js)

```js
let firstBigVariable = 1;
const secondBigVariable = 2;
const thirdBigVariable = 3;
const fourthBigVariable = 4;
const newVeryVeryBigVariable = firstBigVariable && secondBigVariable 
&& thirdBigVariable && fourthBigVariable;

if (newVeryVeryBigVariable) {
  firstBigVariable = 2;
}

```
After committing badCode.js file, eslint is formatting the file into ->  

```js
let firstBigVariable = 1;
const secondBigVarialble = 2;
const thirdBigVariable = 3;
const fourthBigVariable = 4;
const newVeryVeryBigVariable = firstBigVariable && secondBigVarialble && thirdBigVariable && fourthBigVariable;

if (newVeryVeryBigVariable) {
  firstBigVariable = 2;
}

```
### we can see that there is 'max-len' error on line 5 yet the file is committed, 'max-len' error will be thrown while committing this file again.
On next commit -> following issue will be thrown
"5:1  error  Line 5 exceeds the maximum line length of 100  max-len"

**What did you expect to happen?**
 Either it should give 'max-len' error while committing the file for the first time or a new rule should be introduces to handle this scenario.

**What actually happened? Please copy-paste the actual, raw output from ESLint.**
File get commited in git.
Commands: 
git add badCode.js
git commit -m " max-len test commit"

result: File with max-len error gets commited

**Steps to reproduce this issue:**

1. Create a file badCode.js (given above)
2. Add and commit file into your project.
3.  File will get commit.
4. We can see that line 5 is not following the 'max-len' rule

**Are you willing to submit a pull request to fix this bug?** Yes

**Comment by mdjermanovic** [comment]
Hi @rishabh-42, thanks for the issue!

> 1. Create a file badCode.js (given above)
> 1. Add and commit file into your project.
> 1. File will get commit.
> 1. We can see that line 5 is not following the 'max-len' rule

Please note that ESLint does not have a built-in integration with git. You're probably using some tools to run ESLint on pre-commit.

I copy-pasted the code you provided into a `badCode.js` file. The only error reported by ESLint with your configuration was `no-trailing-spaces` on line 5, as there's a space after `secondBigVariable`.

`eslint --fix` correctly removed that space, and didn't merge lines 5 and 6 into one big line.

Are you maybe running something else (e.g., prettier) on pre-commit, _after_ ESLint? 

If not, can you please do the following:

1. Run `npx eslint badCode.js` and share the result (reported errors).
1. Find which command is used to run ESLint before committing, run that command directly on the command line and share the results (the new content of `badCode.js` and reported errors).

**Comment by rishabh-42** [comment]
'prettier'  was being used in the script earlier after eslint. 

Apologies for the wrong issue report. 
Closing the issue.
